main.py

```diff
@@ -54,371 +54,400 @@
 logger.debug("Finished get_all_currencies")
 
 
-def main():
-    """
-    Sells, adjusts TP and SL according to trailing values
-    and buys new coins
-    """
-    # store config deets
-    tp = config['TRADE_OPTIONS']['TP']
-    sl = config['TRADE_OPTIONS']['SL']
-    enable_tsl = config['TRADE_OPTIONS']['ENABLE_TSL']
-    tsl = config['TRADE_OPTIONS']['TSL']
-    ttp = config['TRADE_OPTIONS']['TTP']
-    pairing = config['TRADE_OPTIONS']['PAIRING']
-    test_mode = config['TRADE_OPTIONS']['TEST']
+def buy():
+    while not globals.stop_threads:
+        logger.debug('Waiting for buy_ready event')
+        globals.buy_ready.wait()
+        logger.debug('buy_ready event triggered')
+        if globals.stop_threads:
+            break
+        # the buy block and logic pass
+        if os.path.isfile('new_listing.json'):
+            logger.debug('loading coin from new_listing.json')
+            announcement_coin = load_order('new_listing.json')
+        else:
+            announcement_coin = False
+
+        global supported_currencies
+        if announcement_coin and \
+                announcement_coin not in order and \
+                announcement_coin not in sold_coins and \
+                announcement_coin not in globals.old_coins:
+            logger.info(f'New announcement detected: {announcement_coin}')
+            if not supported_currencies:
+                supported_currencies = get_all_currencies(single=True)
+            if supported_currencies:
+                if announcement_coin in supported_currencies:
+                    logger.debug("Starting get_last_price")
+                    
+                    # get latest price object
+                    obj = get_last_price(announcement_coin, globals.pairing, False)
```

Comment:
can we call this price for readability

Reply:
`obj`? The very next line does `price = obj.last`

This also isn't one of the things I did. If you look at this commit https://github.com/CyberPunkMetalHead/gateio-crypto-trading-bot-binance-announcements-new-coins/pull/63/commits/c9451393dc52ccac2bc5a39b44cc62e8af015abb it has the changes I actually made to the buy and sell blocks.